spa/src/pages/ObjectProperties.vue

```diff
@@ -244,23 +244,25 @@ methods: {
     if(!this.mallObject){
       if(this.walletBalance >= this.price){
         try{
-          const objectPurchase = await this.$http.post(`/object_instance/buy/`, {
-            id: this.objectId});
+            await this.objectProperties();
             if(!this.price){
-              this.error = 'This object is not for sale!'
+              throw new Error('This object is not for sale!')
             }
             if(this.buyer && this.$store.data.user.username !== this.buyer) {
-              this.error = 'This object is reserved for someone else!';
+              throw new Error('This object is reserved for someone else!');
             }
             if(this.price > this.walletBalance){
-              this.error = "You don't have enough cc's.";
-            }
-            if(objectPurchase.data.status === 'success'){
-              setTimeout(this.emitUpdate, 300);
-              this.success = 'Object purchased!';
-              this.error = '';
+              throw new Error("You don't have enough cc's.");
             }
+            await this.$http.post(`/object_instance/buy/`, {
+            id: this.objectId});
+            this.emitUpdate();
+            await this.objectProperties();
+            this.success = 'Object purchased!';
+            this.error = '';
           } catch(errorResponse: any) {
+            this.success = '';
+            this.error = errorResponse;

```

Comment:
I would not recommend displaying whatever the error is on a production version of an app.  This could potentially display the make-up of the database we don't want users to see, or create frustration with a confusing error display.